pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/TypeHelper.java

```diff
@@ -24,28 +24,55 @@ private TypeHelper() {
      * @return <code>true</code> if type node n is of type clazzName or a subtype of clazzName
      */
     public static boolean isA(final TypeNode n, final String clazzName) {
+        final Class<?> clazz = loadClassWithNodeClassloader(n, clazzName);
+
+        if (clazz != null) {
+            return isA(n, clazz);
+        }
+
+        return clazzName.equals(n.getImage()) || clazzName.endsWith("." + n.getImage());
+    }
+    
+    /**
+     * Checks whether the resolved type of the given {@link TypeNode} n is exactly of the type
+     * given by the clazzName.
+     *
+     * @param n the type node to check
+     * @param clazzName the class name to compare to
+     * @return <code>true</code> if type node n is exactly of type clazzName.
+     */
+    public static boolean isExactlyA(final TypeNode n, final String clazzName) {
+        final Class<?> clazz = loadClassWithNodeClassloader(n, clazzName);
+
+        if (clazz != null) {
+            return n.getType() == clazz;
+        }
+
+        return clazzName.equals(n.getImage()) || clazzName.endsWith("." + n.getImage());
+    }
+    
+    private static Class<?> loadClassWithNodeClassloader(final TypeNode n, final String clazzName) {
         if (n.getType() != null) {
             try {
                 ClassLoader classLoader = n.getType().getClassLoader();
                 if (classLoader == null) {
                     // Using the system classloader then
                     classLoader = ClassLoader.getSystemClassLoader();
                 }
-
+    
                 // If the requested type is in the classpath, using the same classloader should work
-                final Class<?> clazz = classLoader.loadClass(clazzName);
-
-                if (clazz != null) {
-                    return isA(n, clazz);
-                }
+                return classLoader.loadClass(clazzName);
```

Comment:
Does this support array types ? You can't pass a name with pairs of brackets to this method, or even the binary name of an array. Using [`ClassUtils.getClass(classLoader, clazzName)`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/ClassUtils.html#getClass-java.lang.ClassLoader-java.lang.String-) would probably be more robust

Reply:
that's sweet! I didn't knew that. Arrays were never supported, but we could add it now...